src/metrics/basicmetrics/plots.py

```diff
@@ -43,37 +43,50 @@ def get_training(path):
     return training
 
 
-def node_total(eval, estimator=np.mean, ax=None):
+def save_fig(plot, save_path, name):
+    """
+    Save the plot at the location specified by `save_path` if `save_path` is not `None`.
+    """
+    if save_path is not None:
+        fig = plot.get_figure()
+        fig.savefig(save_path + name + ".png")
+
+
+def node_total(eval, estimator=np.mean, ax=None, save_path=None):
     plot = sns.lineplot(
         data=eval[eval["Solution"] == 0], y="Nodes", x="Episode", hue="Heuristic", estimator=estimator, ax=ax
     )
     plot.set(xlabel="Evaluation step", ylabel="Nodes visited", title="Node visited until optimality")
+    save_fig(plot, save_path, "eval_node_visited_optimality")
 
 
-def node_first(eval, estimator=np.mean, ax=None):
+def node_first(eval, estimator=np.mean, ax=None, save_path=None):
     first_solution = eval.loc[
         eval[eval["SolutionFound"] == 1].groupby(["Episode", "Instance", "Heuristic"])["Solution"].idxmin()
```

Comment:
This line causes an error on problems without objective values because "SolutionFound" is not one of the column names. We should try to rewrite this line.